### PR TITLE
Refactor tests to use factories instead of fixtures

### PR DESCRIPTION
**Pull request by dunkOnIT**
TODO: 
- [x] create factories
- [x] refactor validations into a service
- [x] check if we have all validations (just take a look for any validation functions missing from controller
- [x] change validate_update_request to use checker service, get old tests running
- [x] go through all old tests, make sure we have the cases covered (add them to a spreadsheet?) 
- [x] once cases are covered, check files for commented out lines and puts statements
- [x] push and flag for review :)

### CODE REVIEW

File: Frontend/src/api/registration/post/submit_registration.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import createClient from 'openapi-fetch'
+iport createClient from 'openapi-fetch'
```
Comment (gregorbg) [inline Frontend/src/api/registration/post/submit_registration.ts:1]: !!?

**Comment by dunkOnIT** [comment]
- Need help resolving the lane_factory conflict
- Just pushing the new stuff for now, will remove the RSwag tests in a later PR

File: Frontend/src/api/registration/post/submit_registration.ts (hunk 1 above)
Comment (dunkOnIT) [reply Frontend/src/api/registration/post/submit_registration.ts:1]: LMAO

Comment (dunkOnIT) [reply Frontend/src/api/registration/post/submit_registration.ts:1]: We caught this in the previous review of this PR, and I only changed it in the other branch :skull:

Comment (dunkOnIT) [reply Frontend/src/api/registration/post/submit_registration.ts]: Should be fixed now

**Comment by dunkOnIT** [comment]
- [ ] Rewrite comeptitor_count function to (a) only consider registrations for one competition, (b) use `where` instead of `all` (consider using a nested query here)
- [ ] Make `target_user_id = @request['user_id'].to_s` as an instance variable in the functions, instead of calling it every time

**Review by FinnIckler — APPROVED** [review]
as per call

**Comment by dunkOnIT** [comment]
- [ ] registration_spec.rb should query database to make sure that is_competing status change are persisting - we shouldn't just test the object in memory

**Pull request merged**